### ISSUE TITLE
Espresense

### ISSUE THREAD

**Issue by hermanndererdmann** [esphome/feature-requests#1412]
**Describe the problem you have/What new integration you would like**

**Please describe your use case for this integration and alternatives you've tried:**

This Project might be very interessting when integrated in Esphome.

https://espresense.com/
**Additional context**

**Comment by mloebl** [comment]
I'd be interested in this as well. I just started using Espresense and I am really impressed. With some of the Chinese OEMs starting to make ESP32 based plug in /outlet modules with bluetooth built in, this would make distributing this around a house seamless for many people and a good fit for the other ESPHome abilities like power monitoring and such.

**Comment by travipross** [comment]
Also very interested!

I'm currently considering desiging an ESP-based module to be placed around the home for primarily temperature sensing. Simultaneously I'm interested in exploring ESPresence. Without integration between these two projects it means up to 2x ESPs per room for this to work nicely. Having this integration support would possibly cut the number of devices in half.

**Comment by FM-17** [comment]
I'm hoping to add ESPresence-enabled ESP32's to my home for pet/asset tracking as well as ESPHome-enabled ESP32's for temp + humidity monitoring and PIR motion detection. Using two ESP32's per room for this would be wasteful, so I too would love to see this added.

**Comment by corgan2222** [comment]
An integration into ESPHome would be really nice! 

If you don't want to wait till this happens, I implemented temp + humidity + pir + radar sensor support into ESPresence. 
https://github.com/ESPresense/ESPresense/pull/54
https://github.com/ESPresense/ESPresense/pull/48

coming this week:
- lux sensor

If you need other sensors, just let us know.

**Comment by virtualducttape** [comment]
This would be great! Esp home is wonderful, adding espresence into it would be a killer feature.

**Comment by Robert7997** [comment]
Building a Multisensor with color led notifikation AND espresence would be a dream 😉👍

I will follow this very closely!

**Comment by corgan2222** [comment]
> 
> 
> Building a Multisensor with color led notifikation AND espresence would be a dream 😉👍
> 
> I will follow this very closely!

I'm actually playing around with LED Support for ESPresense.
How would you use the LED? Should the LED be shown in HA as Light Entity, or in which way you want to use it?

**Comment by rpatel3001** [comment]
If anyone is interested, I put together some code for my own use with esphome that borrows some of the filtering scheme from ESPresense. It doesn't do fingerprinting and is only setup for iBeacons, but if you're handy with code you could easily extend it however you like.

https://github.com/rpatel3001/BleDistance

**Comment by murtoz** [comment]
> How would you use the LED? Should the LED be shown in HA as Light Entity

Yes please. HA is my notification engine and has all the logic. Ideally I would be able to add simple effects to the led like you can in esphome so I can flash it to get my attention. But at least on/off/colour/brightness control from HA to make it usable.

**Comment by Hedda** [comment]
+1 for native room presence detection in ESPHome for Home Assistant's [mqtt_room component](https://www.home-assistant.io/components/sensor.mqtt_room/) or similar "esphome_presence" or "esphome_room" component for localized device presence detection using "trackable beacons" and "base stations" in all rooms.

https://www.home-assistant.io/integrations/mqtt_room/

[ESPresense](https://espresense.com/) was a rewrite of [ESP32-mqtt-room](https://jptrsn.github.io/ESP32-mqtt-room) so maybe ESPHome developers would prefer rewriting it again to not use MQTT?

Ex. Home Assistant built-in integration for presence-based lights builds on the [device_tracker](https://www.home-assistant.io/integrations/device_tracker/) and the [person](https://www.home-assistant.io/integrations/person/) integrations:

https://www.home-assistant.io/integrations/device_sun_light_trigger/

https://www.home-assistant.io/integrations/device_tracker/

https://www.home-assistant.io/integrations/person/

**Comment by MRobi1** [comment]
This PR looks amazing! Thank you @wjtje for working on this. Having the ability to do room-level presence while still having access to all the sensors available in ESPHome will be game-changing for many!

**Comment by virtualducttape** [comment]
Agreed, so excited to see this feature added! (And lucky for me, I've been so lazy that I haven't even started setting up the ESPs around my new house yet)

**Comment by danps1** [comment]
Hi,

I created a feature request outlining a way to solve this problem, if you're like to look at https://community.home-assistant.io/t/apple-devices-presence-using-esphome-ble-proxy-espresense-alternative/524019/1

Thanks

Dan
